src/libostree/ostree-repo-pull.c

```diff
@@ -705,15 +710,43 @@ scan_dirtree_object (OtPullData   *pull_data,
       /* Is this a local repo? */
       if (pull_data->remote_repo_local)
         {
-          if (!import_one_local_content_object (pull_data, file_checksum, cancellable, error))
+          if (!import_one_local_content_object (pull_data, pull_data->remote_repo_local,
+                                                file_checksum, cancellable, error))
             return FALSE;
         }
-      else if (!g_hash_table_lookup (pull_data->requested_content, file_checksum))
+      else
```

Comment:
Shouldn't this still be `else if (!g_hash_table_lookup (pull_data->requested_content, file_checksum))`? If the request has already been made somehow, then we should commit to that, no?